### PR TITLE
MODELIX-781 Add confirmation modal to "delete repository" button

### PR DESCRIPTION
**Pull request by mhuster23**
This PR also changes the position of the delete button and adds "delete branch"-buttons for non-master branches.

## To be verified by reviewers

* [ ] Relevant public API members have been documented
* [ ] Documentation related to this PR is complete
  * [ ] Boundary conditions are documented
  * [ ] Exceptions are documented
  * [ ] Nullability is documented if used
* [ ] Touched existing code has been extended with documentation if missing
* [ ] Code is readable
* [ ] New features and fixed bugs are covered by tests

### CODE REVIEW

File: model-server/src/test/kotlin/org/modelix/model/server/handlers/RepositoriesManagerTest.kt [hunk 1]
```diff
@@ -26,6 +26,16 @@ class RepositoriesManagerTest {
         store.close()
     }
 
+    @Test
+    fun `deleted branches are not listed anymore`() = runTest {
```
Comment (languitar) [inline model-server/src/test/kotlin/org/modelix/model/server/handlers/RepositoriesManagerTest.kt:30]: Not sure if the test case name really reflects what is actually tested. Isn't this more in the direction of "deleting the last branch works"?

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt [hunk 2]
```diff
@@ -98,24 +109,37 @@ class RepositoryOverview(private val repoManager: IRepositoriesManager) {
...
         }
     }
 }
+
+internal fun FlowContent.buildDeleteBranchForm(repositoryId: String, branchName: String) {
+    if (branchName == RepositoryId.DEFAULT_BRANCH) return
+    form {
+        deleteButton {
+            name = "delete"
+            formAction = "../v2/repositories/${repositoryId.encodeURLPathPart()}/branches/${branchName.encodeURLPathPart()}"
```
Comment (languitar) [inline model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt:140]: How does it work that the browser doesn't end up in the response of the REST call after submission?

Comment (mhuster23) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt]: fixed it with some javascript

File: model-server/src/test/kotlin/org/modelix/model/server/handlers/RepositoriesManagerTest.kt (hunk 1 above)
Comment (mhuster23) [reply model-server/src/test/kotlin/org/modelix/model/server/handlers/RepositoriesManagerTest.kt]: renamed

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt [hunk 3]
```diff
@@ -31,7 +36,21 @@ class RepositoryOverview(private val repoManager: IRepositoriesManager) {
...
-                    headContent { title("Repositories") }
+                    headContent {
+                        title("Repositories")
+                        script(type = "text/javascript") {
+                            unsafe {
+                                +"""
+                                    function removeBranch(repository, branch) {
+                                        if (confirm('Are you sure you want to delete the branch' + branch +' of repository' +repository + '?')) {
+                                            fetch('../v2/repositories/' + repository + '/branches/' + branch, { method: 'DELETE'})
+                                            .then( _ => location.reload())
```
Comment (languitar) [inline model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt:47]: I hope we never have an error response from the DELETE endpoint. That would probably be invisible to the user now without opening the browser console.

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt [hunk 4]
```diff
@@ -98,24 +125,33 @@ class RepositoryOverview(private val repoManager: IRepositoriesManager) {
...
         +"Explore Latest Version"
     }
 }
 
-fun FlowContent.buildDeleteForm(repositoryId: String) {
+internal fun FlowContent.buildDeleteRepositoryForm(repositoryId: String) {
     form {
         postButton {
             name = "delete"
             formAction = "../v2/repositories/${repositoryId.encodeURLPathPart()}/delete"
```
Comment (languitar) [inline model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt:144]: I think we have the same problem as with the other confirmation modal here: after deleting, the user might see a blank page? Or not?

**Review by languitar — COMMENTED** [review]
Sorry, one more iteration of things I didn't see in the first round.

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt (hunk 4 above)
Comment (mhuster23) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt:144]: The browser stays on the same page, but the user needs a manual reload to see the removed repository. Should I also add the auto-refresh here?

Comment (languitar) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/RepositoryOverview.kt:144]: Not seeing the effect of an action is probably a UX anti-pattern. So yes, a force reload would be good.

**Review by languitar — APPROVED** [review]

**Pull request merged**